R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+#' from a mortAAR life table list
+#'
+#' The proportional relation between adult males and females (=
+#' Masculinity index) is interesting for a number of reasons: (1)
+#' it can point to basic problems in the datasets in that, say,
+#' one sex is grossly over- or underrepresented. (2) it may hint
+#' towards cultural reaons like sex-specific mobility.\cr
+#' Maternal mortality is a basic indicator for the health system of a
+#' given population. Recently, McFadden and Oxenham (2019) have
+#' provided a formula to calculate it from archaeological data.
+#'
+#' The Masculinity index (MI) is defined for juvenile and older
+#' individuals. Note that with a higher mortality rate of adult females,
+#' an MI < 100 does not necesssarily speak for an unbalanced MI in life.\cr
+#' Maternal mortality is calculated according to the formula
+#' provided by McFadden and Oxenham (2019). McFadden and Oxenham show
+#' that with modern data a very high correlation is achieved by only
+#' comparing the absolute numbers of the age group 20 to 24. This has the
+#' additional advantage that for this age group anthropological aging
+#' methods are reasonable exact.
+#'
+#' @param life_table an object of class mortaar_life_table_list.
+#' @param females an object of class mortaar_life_table for females
+#' @param males an object of class mortaar_life_table for males
+#'
+#' @return Output of masculinity index and maternal mortality.
+#' \itemize{
+#'   \item \bold{Masculinity index}.
+#'
+#'                    \eqn{MI = D>=15male * 100 / D>=15female}
+#'
+#'   \item \bold{Maternal mortality}.
+#'
+#'                    \eqn{333.33 * D20-24female / D20-24male - 76.07}
+#'}
+#'
+#' @examples
+#' # Calculate Masculinity index and maternal mortality from real life
+#' # dataset.
+#' library(magrittr)
+#' muen <- muensingen
+#' muen %>% dplyr::select(age) %>% dplyr::arrange(nchar(age), age) %>%
+#' unique()
+#' muen <- muen %>% dplyr::mutate( age = dplyr::case_when(
+#' .$age == ">50"    ~ "50-70", .$age == ">60"    ~ "60-70",
+#' .$age == "10+/-1" ~ "9-12", .$age == "10"     ~ "10-11",
+#' .$age == "7-8"    ~ "7-9", .$age == "9-10"   ~ "9-11",
+#' TRUE ~  .$age))
+#' muen <- muen %>% tidyr::separate(age, c("from", "to")) %>%
+#' transform(from = as.numeric(from), to = as.numeric(to))
+#' muen_prep <- muen %>% prep.life.table(group = "sex",
+#' agebeg = "from", ageend = "to", method = "Standard",
+#' agerange = "excluded")
+#' muen_result <- muen_prep %>% life.table()
+#' lt.sexrelation(muen_result,"female","male")
+#'
+#' @export
+#'
+lt.sexrelation <- function(life_table, females, males) {
+
+  # check if life_table is of class "mortaar_life_table"
+  if (!inherits(life_table,"mortaar_life_table_list"))  {
+    paste0(
+      "The dataset does not seem to be a mortAAR life table list."
+    ) %>%
+      message
+  } else {
+
+    # Masculinity index (MI) for juvenile and older individuals: males * 100 / females
+
+    fem_age <- life_table[females][[1]][[2]] %>% cumsum
```

Comment:
It's better to access values in the tables with names and not indices. When something changes, debugging with indices only is hell, because you have no idea what the number meant in the past.

Reply:
Hopefully, I have succeeded in changing every instance from index to name.